warehouse/tuf/repository.py

```diff
@@ -0,0 +1,497 @@
+            )
+            delegator_metadata = self.bump_role_version(
+                rolename=delegator,
+                role_metadata=delegator_metadata,
+                role_expires=delegator_metadata.signed.expires,
+                key_rolename=None,
+                store=True,
+            )
+            snapshot_metadata = self.snapshot_update_meta(
+                delegator, delegator_metadata.signed.version, snapshot_metadata
+            )
+
+        return snapshot_metadata
+
+    def bump_role_version(
+        self,
+        rolename: str,
+        role_metadata: Metadata,
+        role_expires: datetime,
+        key_rolename: Optional[str] = None,
+        store: Optional[bool] = False,
+    ) -> Metadata:
+        """
+        Bump an specific Role Metadata (ex: delegated target role) version,
+        optionaly stores it in the repository storage.
+
+        Args:
+            rolename: Specific Role name to have the version bumped
+            role_metadata: Specific Role Metadata
+            role_expires: expiration datetime
+            key_rolename: Different signing Key role name
+            store: Optional, Default False, stores in the repository storage
+        """
+        if key_rolename:
+            key_rolename = key_rolename
+        else:
+            key_rolename = rolename
+        role_metadata.signed.expires = role_expires
+        role_metadata.signed.version += 1
+        key_rolename_keys = self.key_backend.get(key_rolename, "private")
+        for key in key_rolename_keys:
+            role_metadata.sign(SSlibSigner(key), append=True)
+
+        if store:
+            self._store(rolename, role_metadata)
+
+        return role_metadata
+
+    def timestamp_bump_version(
+        self,
+        snapshot_version: int,
+        timestamp_expires: datetime,
+        store: bool = False,
+    ) -> Metadata[Timestamp]:
+        """
+        Bump the Timestamp Role Metadata updating the Snapshot version, and
+        optionally stores it in the repository storage.
+
+        Args:
+            snapshot_version: Snapshot version to be updated on Timestamp role
+            timestamp_expires: New Timestamp expiration datetime
+            store: Optional, Default False, stores in the repository storage
+
+        Returns:
+            Timestamp Metadata
+            ``Metadata[Timestamp]``
+        """
+        timestamp_metadata = self.load_role(Timestamp.type)
+        timestamp_metadata.signed.version += 1
+        timestamp_metadata.signed.expires = timestamp_expires
+        timestamp_metadata.signed.snapshot_meta = MetaFile(version=snapshot_version)
+        timestamp_keys = self.key_backend.get(Timestamp.type, "private")
+        for key in timestamp_keys:
+            timestamp_metadata.sign(SSlibSigner(key), append=True)
+
+        if store:
+            self._store(Timestamp.type, timestamp_metadata)
+
+        return timestamp_metadata
+
+    def snapshot_bump_version(
```

Comment:
Looks like this one could just call `bump_role_version`?

Reply:
Same as above, I did it to be more explicit. But I'm open.

Reply:
I think it's different than in `timestamp_bump_version`, because here you don't assign snapshot's `meta` field, so it's pretty much the same functionality as `bump_role_version`. But we don't have to change this now.